### PR TITLE
Support non-x86 builds

### PR DESCRIPTION
**Pull request by jethrogb**

### CODE REVIEW

**Comment by jethrogb** [comment]
@kwakkel1000 I'm trying to add you as a reviewer but it doesn't let me for some reason.

**Comment by jethrogb** [comment]
bors d=kwakkel1000

File: src/lib.rs [hunk 1]
```diff
@@ -46,10 +46,11 @@
 //! [not sufficient on some platforms]: https://ts.data61.csiro.au/projects/TS/cachebleed/
 
 #![no_std]
-#![cfg_attr(all(test, feature = "nightly"), feature(test))]
+#![cfg_attr(all(test, feature = "nightly"), feature(test, bench_black_box))]
```
Comment (MihirLuthra) [inline src/lib.rs:49]: `bench_black_box` was stabilised in 1.66. (But might be good to keep for people trying with an older toolchain)

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "b64-ct"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (MihirLuthra) [inline Cargo.toml]: Need to bump the version again. Last PR made it `0.1.2`.
Will push a commit for that.

**Comment by MihirLuthra** [comment]
I have also added `cargo check` on `aarch64-linux-android` and `armv7-linux-androideabi` for CI tests.

**Comment by MihirLuthra** [comment]
bors try

**Comment by MihirLuthra** [comment]
@kwakkel1000, I have added tests for `arm64` as well. For osx, I have only added build checks. Apparently, osx machines from travis [need extra $](https://docs.travis-ci.com/user/billing-overview/) and I don't see any support for M series chips either.

**Pull request merged**